routes/apps.js

```diff
@@ -1,4 +1,33 @@
-const express = require('express');
-const router = express.Router();
+// ------- this is like and import statement ---
+    const express = require('express'); //---imports the EXPRESS we installed
+    const router = express.Router(); //--- init router from express imported from line 2
+    const dataAppsJson = require('../data/apps'); //--- import data from data/apps.json
 
+/* --------GET all apps info-------------*/
+
+    router.get('/', (req, res) => {// this router will run if no ID is specified
+        res.json(dataAppsJson);
+    });
+
+/* --------GET specific APP with ID------*/ 
+
+    router.get('/:id', (req, res) => {
+        const found = dataAppsJson.some( dataAppsJson => {dataAppsJson.id === parseInt(req.params.id)} ); // returns true when ID matches
+
+        if(found){
+            res.json(dataAppsJson.filter( dataAppsJson => {dataAppsJson.id === parseInt(req.params.id)} ));
+            // function (dataAppsJson) {
+            //     dataAppsJson.id === parseInt(req.params.id)
+            // }
+        } else {
+            res.status(400).json({msg: `NO APPS with id:${req.params.id} FOUND`});
```

Comment:
Excellent work! This isnt required in the assignments, but it's great you're thinking about the negative cases. 

For reference, the HTTP response code for Not Found is typically 404. 
https://www.restapitutorial.com/httpstatuscodes.html

Reply:
Thank you sir.. Btw how are you here? and you are using the slalom account too.. Hmm so when some one fork your repo, you are automatically a colaborator?

Reply:
I was following the tutorial to understand it more...